patches/server/0925-Add-extra-raytracing-api.patch

```diff
@@ -0,0 +1,96 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: PrimordialMoros <mail@moros.me>
+Date: Fri, 29 Jul 2022 04:18:57 +0300
+Subject: [PATCH] Add extra raytracing api
+
+Adds the option to ignore specific blocks (based on their position) when performing a raytrace.
+
+diff --git a/src/main/java/net/minecraft/world/level/BlockGetter.java b/src/main/java/net/minecraft/world/level/BlockGetter.java
+index d1eefa6ef3e9abfe7af4d8310aa64465fa2d5463..bcce89bd15bc182b612a9a372f9197786e3a3af8 100644
+--- a/src/main/java/net/minecraft/world/level/BlockGetter.java
++++ b/src/main/java/net/minecraft/world/level/BlockGetter.java
+@@ -72,9 +72,13 @@ public interface BlockGetter extends LevelHeightAccessor {
+             return BlockHitResult.miss(clipblockstatecontext1.getTo(), Direction.getNearest(vec3d.x, vec3d.y, vec3d.z), new BlockPos(clipblockstatecontext1.getTo()));
+         });
+     }
++    // Paper start
++    default BlockHitResult clip(ClipContext raytrace1, BlockPos blockposition) {
++      return clip(raytrace1, blockposition, java.util.Set.of());
++    }
+ 
+     // CraftBukkit start - moved block handling into separate method for use by Block#rayTrace
+-    default BlockHitResult clip(ClipContext raytrace1, BlockPos blockposition) {
++    default BlockHitResult clip(ClipContext raytrace1, BlockPos blockposition, java.util.Set<BlockPos> ignored) { // Paper end
```

Comment:
One comment block for the `clip(ClipContext,BlockPos)` method,
and then 1 inline comment on the method you added a parameter too